bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -358,8 +358,7 @@ func (r *KubeadmConfigReconciler) reconcile(ctx context.Context, scope *Scope, c
 	}
```

Comment:
Audited for usages of the deprecated v1beta1 condition packages.

These are the results:
* test/e2e: 1x v1beta1conditions.IsFalse + 1x v1beta1conditions.Get directly below
* CAPD:
  * 2x v1beta1conditions.GetReason
  * 1x v1beta1conditions.Has
  * a few times v1beta1conditions.IsTrue
  * a few times v1beta1conditions.Get

I suspect CAPD is expected at this point. Can you please check that this remaining work is properly tracked?

Reply:
If I got it right:
- test/e2e: 1x v1beta1conditions.IsFalse -> MachinePool 😢 
- test/e2e: 1x v1beta1conditions.Get directly below -> It is a check for MS MachinesCreated condition, which doesn't have an equivalent in v1beta2
- I suspect CAPD is expected at this point -> Correct, it is tracked

Reply:
Added a test on new conditions along side the test for MS MachinesCreated condition